ethereum/spec/src/main/java/tech/pegasys/teku/spec/SpecVersion.java

```diff
@@ -36,6 +37,17 @@ private SpecVersion(
     this.schemaDefinitions = schemaDefinitions;
   }
 
+  public static SpecVersion createForFork(final Bytes4 fork, final SpecConfig specConfig) {
+    if (specConfig.getGenesisForkVersion().equals(fork)) {
+      return createPhase0(specConfig);
+    } else if (specConfig.toVersionAltair().isPresent()
+        && SpecConfigAltair.required(specConfig).getAltairForkVersion().equals(fork)) {
```

Comment:
Not sure this is better, but maybe we could do something like:
```suggestion
    } else if (specConfig.altair(SpecConfigAltair::getAltairForkVersion).map(v -> v.equals(fork)).orElse(false)) {
```

Reply:
hmm, not sure about the `specConfig.altair` method.  I'm really hoping that all the Altair specific logic can just be given a `SpecConfigAltair` and not have to deal with `Optional` all the time.  Otherwise the code just gets littered with `.orElseThrow`.

I have changed this to use the `toVersionAltair().map` approach though as it probably is a bit neater.